packages/nestjs/src/guards/Auth.guard.ts

```diff
@@ -0,0 +1,74 @@
+import { authenticated, User } from '@accounts/graphql-api';
+import { CanActivate, ExecutionContext, Injectable } from '@nestjs/common';
+import { Reflector } from '@nestjs/core';
+import { GqlExecutionContext } from '@nestjs/graphql';
+import { AuthValidatorFn } from '../decorators/AuthValidator.decorator';
+import { AccountsSessionRequest } from '../interfaces/AccountsRequest';
+import { AUTH_VALIDATOR_FUNCTIONS } from '../utils/accounts.constants';
+import { GQLParam } from '../utils/GraphQLUtils';
+
+// todo: unify context better. It sounds like this will break see issue: https://github.com/nestjs/nest/issues/1581 and PR https://github.com/nestjs/nest/pull/2493
+@Injectable()
+export class AuthGuard implements CanActivate {
+  constructor(private readonly reflector: Reflector) {}
+
+  async canActivate(context: ExecutionContext): Promise<boolean> {
+    const ctx = GqlExecutionContext.create(context);
+    const gqlCtx = ctx.getContext();
+    if (gqlCtx) {
+      try {
+        const gqlParams: GQLParam = [ctx.getRoot(), ctx.getArgs(), gqlCtx, ctx.getInfo()];
+        // use the authenticated function from accounts-js. All that's really needed is context
+        await authenticated(() => null)(...gqlParams);
+        return this.runValidators(gqlCtx.user, context, gqlParams);
```

Comment:
is it `gqlCtx.req.user`?

Reply:
yea, I think you're right

Reply:
How does this test pass tho? Does accounts set

Test cases: https://github.com/NickBolles/accounts/blob/njb/nest-module/packages/nestjs/examples/with-auth-guard/e2e/app.e2e-spec.ts#L220-L282

Resolver: https://github.com/NickBolles/accounts/blob/njb/nest-module/packages/nestjs/examples/with-auth-guard/DarthsResolver.ts

Reply:
Actually, no what I have is correct. This is where Accounts.js builds the context, it sets user, as well as some more properties on the context.

https://github.com/accounts-js/accounts/blob/master/packages/graphql-api/src/utils/context-builder.ts#L37-L43

Interestingly, it doesn't set the request on the context, but it looks like it requires it to be set on the context by something else earlier on